internal/blockchain/ethereum/ethereum.go

```diff
@@ -947,3 +947,66 @@ func (e *Ethereum) GetAndConvertDeprecatedContractConfig(ctx context.Context) (l
 	})
 	return location, fromBlock, err
 }
+
+func mergeObjects(detail interface{}, output fftypes.JSONObject) (fftypes.JSONObject, error) {
+
+	var mergedObj map[string]interface{}
+
+	outputJSON, err := json.Marshal(output)
+
+	if err != nil {
+		return output, err
+	}
+
+	detailJSON, err := json.Marshal(detail)
+
+	if err != nil {
+		return output, err
+	}
+
+	err = json.Unmarshal(outputJSON, &mergedObj)
+
+	if err != nil {
+		return output, err
+	}
+
+	err = json.Unmarshal(detailJSON, &mergedObj)
+
+	if err != nil {
+		return output, err
+	}
+
+	return fftypes.JSONObject(mergedObj), nil
+}
+
+func (e *Ethereum) GetTransactionStatus(ctx context.Context, operation *core.Operation) (interface{}, error) {
+	txnID := (&core.PreparedOperation{ID: operation.ID, Namespace: operation.Namespace}).NamespacedIDString()
+	transactionRequestPath := fmt.Sprintf("/transactions/%s", txnID)
+	client := e.client
+	if client == nil {
+		client = e.client
+	}
+	var resErr ethError
+	var statusResponse interface{}
+	res, err := client.R().
+		SetContext(ctx).
+		SetError(&resErr).
+		SetResult(&statusResponse).
+		Get(transactionRequestPath)
+	if err != nil || !res.IsSuccess() {
+		return nil, wrapError(ctx, &resErr, res, err)
+	}
+
+	if operation.Status == core.OpStatusPending {
+		// handleReceipt() expects a single object with the fields from both operation.Output
+		// and the blockchain connector's transaction detail, so we merge the two
+		// before passing to handleReceipt()
+		mergedObj, err := mergeObjects(statusResponse, operation.Output)
+		if err != nil {
+			// Log a warning and return the output on its own
+			log.L(ctx).Warnf("Failed to add detail to transaction status: %v", err)
+		}
+		e.handleReceipt(ctx, mergedObj)
```

Comment:
Would like to understand this a bit better.
Is this something that's historical from another part of the code?

... digging a bit myself to find out

Reply:
No, the `handleReceipt` function seems to directly take the output from a WebSocket notification as far as I can tell:
https://github.com/hyperledger/firefly/blob/8f380025acc10713a437a0400973df5ac393b59f/internal/blockchain/ethereum/ethereum.go#L348-L371

Those notifications are of a different form to just the `/transaction/{id}` data. They contain a `headers` section (which is inherited from the EthConnect API that came before).
It's a very simple wrapper as so:
https://github.com/hyperledger/firefly-transaction-manager/blob/affb781dc00216703b2b3945cff556b8797e0f93/pkg/fftm/policyloop.go#L288-L305

Reply:
I've done some refactoring under the new commit https://github.com/hyperledger/firefly/pull/1111/commits/547432920ab2e39aad1a281f9f917a73d0c05e24 which I think addresses the other comments. Let me know if that looks more like what you were expecting re. what we pass to `handleReceipt()`

Might be worth a chat about https://github.com/hyperledger/firefly/pull/1111#discussion_r1039628753 to discuss in a bit more detail?